ipatests/test_integration/test_ipahealthcheck.py

```diff
@@ -377,21 +377,18 @@ def test_ipa_healthcheck_fips_enabled(self):
                                             failures_only=False)
         assert returncode == 0
 
-        cmd = self.master.run_command(
-            [paths.FIPS_MODE_SETUP, "--is-enabled"], raiseonerr=False
-        )
-        returncode = cmd.returncode
+        is_fips_enabled = tasks.is_fips_enabled(self.master)
 
         assert "fips" in check[0]["kw"]
 
         if check[0]["kw"]["fips"] == "disabled":
-            assert returncode == 2
+            assert not is_fips_enabled
         elif check[0]["kw"]["fips"] == "enabled":
-            assert returncode == 0
-        elif check[0]["kw"]["fips"] == f"missing {paths.FIPS_MODE_SETUP}":
-            assert returncode == 127
+            assert is_fips_enabled
         else:
-            assert returncode == 1
+            raise ValueError("File %s doesn't exist or contains unexpected "
+                             "value, this is a kernel issue!"
+                             % paths.PROC_FIPS_ENABLED)
```

Comment:
In general the patch LGTM, but it needs to take into account the ipa-healthcheck version.
For instance:
- RHEL 10.0 with ipa-healthcheck 0.16-7
- ipa-healthcheck does not have the patch from https://github.com/freeipa/freeipa-healthcheck/pull/349/ and returns "fips": "missing /bin/fips-mode-setup"
- the test needs to handle that case and add a xfail based on ipa-healthcheck version + os version
This way we will be able to get the patch in RHEL 10.0 0day and have a green test thanks to the xfail annotation.

Reply:
Would it be simpler to see if fips-mode-setup exists on the target host and xfail if it does not?
It would probably be best I suppose if I cut a new release of healthcheck but it'll take a while to trickle out.